org.eclipse.dawnsci.analysis.api/src/org/eclipse/dawnsci/analysis/api/tree/TreeUtils.java

```diff
@@ -66,8 +67,22 @@ private static String getPathDepthFirst(final String parent, final GroupNode gro
 	 * @return map with full path as key and NodeLink as value
 	 */
 	public static Map<String, NodeLink> treeBreadthFirstSearch(GroupNode file, IFindInTree finder, boolean findFirst, IMonitor mon) {
+		return treeBreadthFirstSearch(file, finder, findFirst, false, mon);
+	}
+	
+	/**
+	 * Breadth first search of nexus tree using visitor to determine whether a node link matches
+	 * those searched for
+	 * @param file
+	 * @param finder
+	 * @param findFirst
+	 * @param maintainOrder
+	 * @param mon
+	 * @return map with full path as key and NodeLink as value
+	 */
+	public static Map<String, NodeLink> treeBreadthFirstSearch(GroupNode file, IFindInTree finder, boolean findFirst, boolean maintainOrder, IMonitor mon) {
 		
-		Map<String, NodeLink> map = new HashMap<String, NodeLink>();
+		Map<String, NodeLink> map = maintainOrder ? new LinkedHashMap<String, NodeLink>() :  new HashMap<String, NodeLink>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)